client/driver/docker.go

```diff
@@ -489,6 +497,25 @@ func (d *DockerDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle
 	return h, nil
 }
 
+func (d *DockerDriver) Cleanup(_ *ExecContext, res *CreatedResources) {
```

Comment:
Refactored Cleanup to take a single resource key/value pair at a time and let TaskRunner handle iterating over CreatedResources, tracking failues, and retrying.